src/main/scala/com/lunatech/lunabot/LunabotServlet.scala

```diff
@@ -36,39 +36,40 @@ class LunabotServlet extends ScalatraServlet with JacksonJsonSupport {
     logger.debug(request.body)
     val jsonValue = parse(request.body.replace("mention_name", "mentionName"))
     val hipchatMsg = jsonValue.extract[HipChatMessage]
+    val cmd: String = hipchatMsg.item.message.message.replace("/scala ", "")
     val authToken = System.getenv("AUTH_TOKEN")
-    val urlStr = "https://api.hipchat.com/v2/room/roomId/notification?auth_token="
 
-    val executionFuture: Future[String] = Future {
-      val cmd1: String = hipchatMsg.item.message.message.replace("/scala", "")
-      val output: Seq[String] = Seq("scala", "-e", cmd1)
-      output.!!
-    }
 
-    executionFuture.onComplete {
+    //Send response to HipChat Room
+    executeExpression(cmd).onComplete {
       case Success(result) => {
-        //Send response to HipChat Room
-        val myRequest = sendAsyncResponse(urlStr, hipchatMsg, authToken) <<
+        val myRequest = prepareResponse(hipchatMsg, authToken) <<
           """{"color": "green", "message": """ + "\"" + "@" + hipchatMsg.item.message.from.mentionName + " " +
-            result.replace("\n", "\\n") + "\"" + """, "notify": false, "message_format": "text"}"""
+            result.replace("\n", "\\n") + "\"" +
+            """, "notify": false, "message_format": "text"}"""
         dispatch.Http(myRequest)
       }
       case Failure(ex: Exception) => {
-        val myRequest = sendAsyncResponse(urlStr, hipchatMsg, authToken) <<
+        val myRequest = prepareResponse(hipchatMsg, authToken) <<
           s"""{"color": "red", "message": "Sorry your scala code has failed with ${ex.getMessage}
              |", "notify": false, "message_format": "text"}""".stripMargin
         dispatch.Http(myRequest)
       }
 
     }
-
   }
 
-  def sendAsyncResponse(urlStr: String, hipchatMsg: HipChatMessage, authToken: String): Req = {
-    val roomId = hipchatMsg.item.room.id
-    val updatedUrl = urlStr.replace("roomId", roomId.toString).concat(authToken)
-    val request = dispatch.url(updatedUrl).setContentType("application/json", "UTF-8").POST
-    request
-  }
-}
 
+    def prepareResponse(hipchatMsg: HipChatMessage, authToken: String): Req = {
+      val roomId = hipchatMsg.item.room.id
+      val urlStr = s"https://api.hipchat.com/v2/room/${roomId}/notification?auth_token="+authToken
+      val request = dispatch.url(urlStr).setContentType("application/json", "UTF-8").POST
+      request
+    }
+
+    def executeExpression(scalaExpr: String): Future[String] = Future {
+      val output: Seq[String] = Seq("scala", "-e", scalaExpr)
+      output.!!
+    }
+
+}
```

Comment:
New line here needed. Can you please fix your editor settings to add new lines at the end ?